scheds/rust/scx_p2dq/src/bpf/main.bpf.c

```diff
@@ -950,31 +955,35 @@ void BPF_STRUCT_OPS(p2dq_stopping, struct task_struct *p, bool runnable)
 	trace("%s weight %d slice %llu used %llu scaled %llu",
 	      p->comm, p->scx.weight, last_dsq_slice_ns, used, scaled_used);
 
-	// On stopping determine if the task can move to a longer DSQ by
-	// comparing the used time to the scaled DSQ slice.
-	if (used >= ((9 * last_dsq_slice_ns) / 10)) {
-		if (taskc->dsq_index < nr_dsqs_per_llc - 1) {
-			taskc->dsq_index += 1;
-			stat_inc(P2DQ_STAT_DSQ_CHANGE);
-			trace("%s[%p]: DSQ inc %llu -> %u", p->comm, p,
-			      taskc->last_dsq_index, taskc->dsq_index);
-		} else {
-			stat_inc(P2DQ_STAT_DSQ_SAME);
-		}
-	// If under half the slice was consumed move the task back down.
-	} else if (used < last_dsq_slice_ns / 2) {
-		if (taskc->dsq_index > 0) {
-			taskc->dsq_index -= 1;
-			stat_inc(P2DQ_STAT_DSQ_CHANGE);
-			trace("%s[%p]: DSQ dec %llu -> %u", p->comm, p,
-			      taskc->last_dsq_index, taskc->dsq_index);
+	if (!runnable) {
```

Comment:
And move this logic to ops.quiescent()? In this case it should be equivalent IIUC, but maybe moving the logic to separate callbacks make the code easier to follow?